app/controllers/three_d_secure_controller.js

```diff
@@ -31,7 +31,7 @@ const build3dsPayload = (chargeId, req) => {
   const paRes = _.get(req, 'body.PaRes')
   if (!_.isUndefined(paRes)) {
     auth3dsPayload.pa_response = paRes
-    logger.info(`paRes received for charge [${chargeId}] 3DS authorisation [${paRes}]`)
+      logger.info(`paRes received for charge [${chargeId}] 3DS authorisation [starts with [${paRes.substring(0,50)}] and ending with [${paRes.substring(paRes.length-50)}] ]`)
```

Comment:
What if length is < 50?
(when argument passed to substring is < 0 it will be treated as 0 - is it desired behaviour?)

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring

Reply:
yes, for the log lines we are interested in paRes with length > 4000 which are failing 3DS payments